### PR TITLE
packaging

### PR DESCRIPTION
**Pull request by JonathanLehner**
I fetched from upstream and merged and tested with the packaging test colab notebooks
here you go:
https://colab.research.google.com/gist/JonathanLehner/dc8848488598af2ac4df7251cf7f6a59/gpu-test.ipynb
https://colab.research.google.com/gist/JonathanLehner/6d2342fb677ce23b2e4cd5428c7045c8/vid2cleantxt-cpu-test.ipynb

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -100,14 +102,21 @@ Essentially, clone the repo, and run `python vid2cleantxt/transcribe.py --input-
...
+    -   download the video with `python examples/TEST_singlefile/dl_src_video.py`
 
 -   if you are new, you can clone with [github desktop](https://desktop.github.com/)
 -   if neither option works for you, check out the Colab notebooks distributed with this repo.
 
 > \* the example videos need to be downloaded with the scripts in the relevant dirs, such as `python examples/TEST_singlefile/dl_src_video.py`
 
+## Use as package
+1.  `pip install .`
+2.  `spacy download en_core_web_sm`
```
Comment (pszemraj) [inline README.md:116]: does the standard way of getting the package not work? do you have reproducible code? going to be kind of awkward to get this on pypi and force downloading the package manually.

I'd like to at least try it and see it not work before changing the way it's downloaded

**Comment by pszemraj** [comment]
ok so thus far I checked the GPU test and that works 👍 I had some issues when I directly wanted to go with cloning and then `pip install .` (with no reqs installed first) so I am going to investigate that further to see if I did something wrong or not

**Comment by pszemraj** [comment]
let's fix the spacy model package issue with [one of the solutions listed here](https://stackoverflow.com/questions/53383352/spacy-and-spacy-models-in-setup-py) and then I think we can finally merge. I can provide a code suggestion sometime later but something like this should work:

```
import spacy

def load_spacy_models():
    """downloads spaCy models if not installed on local machine."""
    try:
        nlp = spacy.load('en')
    except OSError:
        print('INFO: Downloading language model for the spaCy POS tagger\n'
            "(don't worry, this will only happen once)", file=stderr)
        from spacy.cli import download
        download('en')
        nlp = spacy.load('en')

```

and then `vid2cleantxt/transcribe.py` would call `load_spacy_models()` at the start of `main()`.

**Comment by pszemraj** [comment]
Of course, if we can get it to work inside `setup.py` reliably, that would be ideal. Also open to any other/better solutions

**Comment by pszemraj** [comment]
oooook sorry for the delays. per [this test notebook](https://colab.research.google.com/gist/pszemraj/f2be9c7df09e346f185413dc1b5a6221/cpu-test-for-jonathanlehner-vid2cleantxt.ipynb) the issue may have fixed itself in the interim, but let's add the below to `./vid2cleantxt/v2ct_utils.py` just in case.

```python
import spacy
from spacy.cli import download

def load_spacy_models():
    """downloads spaCy models if not installed on local machine."""
    try:
        nlp = spacy.load("en")
    except OSError as e:
        print(
            "INFO: Downloading language model for the spaCy POS tagger\n"
            "(don't worry, this will only happen once)",
        )
        logging.info(f"downloading the spacy model en_core_web_sm due to:\t{e}")
        download("en_core_web_sm")
        nlp = spacy.load("en_core_web_sm")
```

then in line 49 of `./vid2cleantxt/vid2cleantxt/transcribe.py` 

```python
from vid2cleantxt.v2ct_utils import load_spacy_models
load_spacy_models()
```

**Review by JonathanLehner — COMMENTED** [review]
looks good

**Comment by JonathanLehner** [comment]
you need to approve 
image

**Review by pszemraj — APPROVED** [review]
thanks for updates !!

**Pull request merged**